### DISCUSSION TITLE
RNP integration with gopass

### DISCUSSION THREAD

**Discussion by shemeshg** [gopasspw/gopass#2776] [Ideas]
Hello gopass community,

I am interested in the possibility of integrating RNP, a high-performance OpenPGP library, with gopass, the slightly more awesome standard unix password manager for teams. I have been using RNP for some time and I find it very fast, reliable, and secure. I think it would be a great addition to gopass, especially for Windows users who want a native OpenPGP implementation.

I have some questions about this potential integration:

Is it planned or in progress? If so, what is the current status and roadmap?
Would it be implemented as a thread or as a process (like the current age/gnupg options)?

I think this is a good suggestion because it would enable multithreading for search operations, which would improve the performance and user experience of gopass. 
It would also make the gopass implementation of RNP fully native on Windows x64, which would reduce the dependency on external tools and simplify the installation and setup process.

I would appreciate any feedback or comments on this idea. 

Thanks, `gopass` is awesome

also:
* Will it work in terms of compatibility with other pass implementations and existing gopass features (such as mounts, 
   -YAML support [maybe yaml-cpp] 
   -(It seems gopass already know how to hook a password for `age`, so it might be used for rnpgp implmentation)

**Comment by dominikschulz** [comment]
RNP is C++ only, isn't it? Currently we will not accept any CGo dependencies into gopass since that would break our entire build and test stack.

Calling into an external process is acceptable, but then we would need to discuss the value of RNP. I don't think performance is a concern currently (but feel free to disagree).

**Comment by shemeshg** [comment]
[RNP has a CLI tool, but its installation is currently broken due to the Botan 2.x to 3.x migration However, a new version is expected to be released this month.

though for my experience RNP is four times faster than GPG in single-thread mode, and that advantage is multiplied by the number of CPUs. Unless we consider the performance improvement and multi-threading support of RNP, there is not much benefit in integrating it.
